src/main/java/com/squareup/javapoet/JavaxLangModelPoet.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (C) 2015 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.javapoet;
+
+import static com.squareup.javapoet.Util.checkNotNull;
+
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.TypeParameterElement;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.DeclaredType;
+import javax.lang.model.type.ExecutableType;
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.type.TypeVariable;
+import javax.lang.model.util.Elements;
+import javax.lang.model.util.Types;
+
+/**
+ * Provides utilities based on parsing elements and types from javax.lang.model instances.
+ *
+ * @author Christian Stein
+ */
+class JavaxLangModelPoet {
+
+  protected final Elements elements;
+  protected final Types types;
+
+  JavaxLangModelPoet(Elements elements, Types types) {
+    checkNotNull(elements, "elements==null");
+    checkNotNull(types, "types==null");
+    this.elements = elements;
+    this.types = types;
+  }
+
+  /**
+   * Create a method spec builder that copied 1:1 from {@code method} and is viewed as being a
+   * member of the specified {@code containing} class or interface.
+   *
+   * This will copy its modifiers, annotations, type parameters, return type, name, parameters and
+   * throws declarations.
+   */
+  public MethodSpec.Builder builder(ExecutableElement method, DeclaredType containing) {
```

Comment:
After the merge of https://github.com/square/javapoet/pull/287 - what to do with this feature of creating an 1:1 copy from an `ExecutableElement`?